src/rustdoc-json-types/lib.rs

```diff
@@ -300,24 +303,68 @@ pub struct Struct {
 pub struct Enum {
     pub generics: Generics,
     pub variants_stripped: bool,
+    /// Will all be [`ItemEnum::Variant`]
     pub variants: Vec<Id>,
     pub impls: Vec<Id>,
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
-#[serde(rename_all = "snake_case")]
-#[serde(tag = "variant_kind", content = "variant_inner")]
-pub enum Variant {
-    Plain,
-    Tuple(Vec<Type>),
-    Struct(Vec<Id>),
+pub struct Variant {
+    pub kind: StructKind,
```

Comment:
I think we need a separate enum for variant kinds and struct kinds, as unit variants may have a discriminant value, which currently we don't document but should. See https://github.com/Enselic/cargo-public-api/issues/121 and [zulip discussion](https://rust-lang.zulipchat.com/#narrow/stream/266220-rustdoc/topic/Enum.20discriminant.20values.20in.20HTML.20output)